Model/src/main/java/org/eupathdb/sitesearch/wsfplugin/CommunityStudyIdsPlugin.java

```diff
@@ -0,0 +1,131 @@
+import org.gusdb.wdk.model.question.Question;
+import org.gusdb.wdk.model.record.PrimaryKeyDefinition;
+import org.gusdb.wdk.model.record.RecordClass;
+import org.gusdb.wdk.model.user.User;
+import org.gusdb.wsf.plugin.AbstractPlugin;
+import org.gusdb.wsf.plugin.PluginModelException;
+import org.gusdb.wsf.plugin.PluginRequest;
+import org.gusdb.wsf.plugin.PluginResponse;
+import org.gusdb.wsf.plugin.PluginUserException;
+import org.gusdb.fgputil.db.runner.SQLRunner;
+import org.gusdb.fgputil.db.runner.SQLRunnerException;
+import org.gusdb.wdk.model.user.UserFactory;
+
+import org.gusdb.fgputil.runtime.GusHome;
+import org.gusdb.fgputil.runtime.InstanceManager;
+import org.gusdb.wdk.model.Utilities;
+import org.gusdb.wdk.model.WdkModel;
+import org.apache.log4j.Logger;
+
+import javax.sql.DataSource;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class CommunityStudyIdsPlugin extends AbstractPlugin {
+
+    private static final Logger LOG = Logger.getLogger(CommunityStudyIdsPlugin.class);
+
+    static final String PROJECT_ID_PROPLIST = "projectId";  // this is a <propertyList> used in the model xml
+    static final String VDI_SCHEMA_SUFFIX_PROP_KEY = "VDI_SCHEMA_SUFFIX";  // this is in model.prop
+    static final String OAUTH_SERVICE_URL_PROP_KEY = "OAUTH_SERVICE_URL";  // this is in model.prop
+
+    @Override
+    public String[] getRequiredParameterNames() {
+        return new String[0];
+    }
+
+    @Override
+    public String[] getColumns(PluginRequest request) throws PluginModelException {
+        RecordClass recordClass = getQuestion(request).getRecordClass();
+        PrimaryKeyDefinition pkDef = recordClass.getPrimaryKeyDefinition();
+        String[] dynamicColumns =  new String[]{ "owner_name", "owner_institution" };
+        String[] columns = ArrayUtil.concatenate(pkDef.getColumnRefs(), dynamicColumns);
+        LOG.info("CommunityStudyIdsPlugin instance will return the following columns: " + FormatUtil.join(columns, ", "));
+        return columns;
+    }
+
+    /*
+     - query appDB VDI control tables to get map from owner's user_id to dataset id. store in memory.
+     - query (REST) oauth service to get user_id:details mapping, for the owners of community datasets.  store in memory
+     - merge these two, to produce (dataset_id, owner_name, owner_institution)
+    */
+    @Override
+    protected int execute(PluginRequest request, PluginResponse response) throws PluginModelException, PluginUserException {
+        Question question = getQuestion(request);
+        WdkModel wdkModel = question.getRecordClass().getWdkModel();
+        List<UserDatasetIds> communityDatasetIds = getCommunityDatasetIds(request, question, wdkModel);
+        UserFactory userFactory = new UserFactory(wdkModel);
+        List<Long> userIds = communityDatasetIds.stream().map(udi -> udi.ownerId).collect(Collectors.toList());
```

Comment:
Probably want to collect into a set here.  The method in UserFactory will do it too but maybe be clear that you want to remove duplicates?  Just use Collectors.toSet().